src/commands/kubeconfig.ts

```diff
@@ -0,0 +1,248 @@
+          type: "string",
+          demandOption: true,
+          describe: "The ID of the k8s cluster as configured P0 Security",
+        })
+        .option("resource", {
+          type: "string",
+          describe:
+            'The resource or resource type (e.g., "Pod / *"), or omit for all',
+        })
+        .option("role", {
+          type: "string",
+          demandOption: true,
+          describe:
+            'The k8s role to request, e.g., "ClusterRole / cluster-admin"',
+        })
+        .option("reason", {
+          type: "string",
+          describe: "Reason access is needed",
+        })
+        .option("requested-duration", {
+          type: "string",
+          // Copied from the P0 backend
+          describe:
+            "Requested duration for access (format like '10 minutes', '2 hours', '5 days', or '1 week')",
+        }),
+    guard(kubeconfigAction)
+  );
+
+const kubeconfigAction = async (
+  args: yargs.ArgumentsCamelCase<KubeconfigCommandArgs>
+) => {
+  if (!(await ensureEksInstall())) {
+    throw "Required dependencies are missing; please try again after installing them, or check that they are available on the PATH.";
+  }
+
+  const role = normalizeRoleArg(args.role);
+
+  if (args.resource) {
+    validateResourceArg(args.resource);
+  }
+
+  const authn = await authenticate();
+
+  const clusterConfig = await getAndValidateK8sIntegration(authn, args.cluster);
+
+  const { clusterId, awsAccountId, awsClusterArn } = clusterConfig;
+  const { config } = await getAwsConfig(authn, awsAccountId);
+  const { login } = config;
+
+  // Verify that the AWS auth type is supported before issuing the requests
+  if (!login?.type || login?.type === "iam") {
+    throw "This AWS account is not configured for kubectl access via the P0 CLI.\nYou can request access to the cluster using the `p0 request k8s` command.";
+  }
+
+  const { type: loginType } = login;
+
+  const request = await requestAccessToCluster(authn, args, clusterId, role);
+
+  const awsAuth = await awsCloudAuth(
+    authn,
+    awsAccountId,
+    request.generated,
+    loginType
+  );
+
+  const profile = profileName(clusterId);
+
+  // The `aws eks update-kubeconfig` command can't handle the ARN of the EKS cluster.
+  // So we must, with great annoyance, parse it to extract the cluster name and region.
+  const clusterInfo = extractClusterNameAndRegion(awsClusterArn);
+  const { clusterRegion, clusterName } = clusterInfo;
+
+  await writeAwsTempCredentials(profile, awsAuth);
+  await writeAwsConfigProfile(profile, { region: clusterRegion });
+
+  const updateKubeconfigArgs = [
+    "eks",
+    "update-kubeconfig",
+    "--name",
+    clusterName,
+    "--region",
+    clusterRegion,
+    "--profile",
+    profile,
+  ];
+
+  try {
+    // Federated access especially sometimes takes some time to propagate, so
+    // retry for up to 20 seconds just in case it takes a while.
+    const awsResult = await retryWithSleep(
```

Comment:
Hmm, it might be? But it seems to have a lot of SSH-specific behavior, so I'd want to make a more general version to apply here. But it also seems like it might be a bit overkill in this instance as well? `accessPropagationGuard()` looks to be doing quite a bit of checking for specific, SSH-related errors, when really the only issue here is that sometimes access is denied while the permissions propagate, especially in the case of Okta SAML.